File: src/main/kotlin/stats/StatsInstall.kt
package com.tbread.stats

import com.tbread.config.PropertyHandler
import java.util.UUID

object StatsInstall {
    private const val KEY_INSTALL_ID = "statsInstallId"

    fun installId(): String {
        val saved = PropertyHandler.getProperty(KEY_INSTALL_ID)?.takeIf { it.isNotBlank() }
        if (saved != null) return saved
        val generated = UUID.randomUUID().toString()
        PropertyHandler.setProperty(KEY_INSTALL_ID, generated)
        return generated
    }
}

File: dotnet/src/WaffleMeter.Stats/StatsInstall.cs
using WaffleMeter.Services;

namespace WaffleMeter.Stats;

/// <summary>
/// Verbatim port of Kotlin <c>stats.StatsInstall</c>: a stable per-install UUID persisted in
/// settings (generated on first use). Sent as the <c>x-install-id</c> header.
/// </summary>
public static class StatsInstall
{
    private const string KeyInstallId = "statsInstallId";

    public static string InstallId(PropertyHandler properties)
    {
        string? saved = properties.GetProperty(KeyInstallId);
        if (!string.IsNullOrWhiteSpace(saved))
        {
            return saved!;
        }

        string generated = Guid.NewGuid().ToString();
        properties.SetProperty(KeyInstallId, generated);
        return generated;
    }
}
